worlds/spelunker/Client.py

```diff
@@ -0,0 +1,207 @@
+import logging
+import struct
+import time
+from struct import pack
+from .Rom import location_table, hidden_table
+from typing import TYPE_CHECKING, Dict, Set
+
+# TODO:  Remove this when Archipelago 0.4.4 gets released
+import sys
+
+if "worlds._bizhawk" not in sys.modules:
+    import importlib
+    import os
+    import zipimport
+
+    bh_apworld_path = os.path.join(
+        os.path.dirname(sys.modules["worlds"].__file__), "_bizhawk.apworld"
+    )
+    if os.path.isfile(bh_apworld_path):
+        importer = zipimport.zipimporter(bh_apworld_path)
+        spec = importer.find_spec(os.path.basename(bh_apworld_path).rsplit(".", 1)[0])
+        mod = importlib.util.module_from_spec(spec)
+        mod.__package__ = f"worlds.{mod.__package__}"
+        mod.__name__ = f"worlds.{mod.__name__}"
+        sys.modules[mod.__name__] = mod
+        importer.exec_module(mod)
+    elif not os.path.isdir(os.path.splitext(bh_apworld_path)[0]):
+        raise AssertionError("Could not import worlds._bizhawk")
+
+from NetUtils import ClientStatus
+import worlds._bizhawk as bizhawk
+from worlds._bizhawk.client import BizHawkClient
+import time
+
+if TYPE_CHECKING:
+    from worlds._bizhawk.context import BizHawkClientContext
+else:
+    BizHawkClientContext = object
+
+# Add .apwl suffix to bizhawk client
+from worlds.LauncherComponents import SuffixIdentifier, components
+
+for component in components:
+    if component.script_name == "BizHawkClient":
+        component.file_identifier = SuffixIdentifier(
+            *(*component.file_identifier.suffixes, ".apsplunker")
+        )
+        break
+
+EXPECTED_ROM_NAME = "SPELUNKERAP"
+
+
+class SpelunkerClient(BizHawkClient):
+    game = "Spelunker"
+    system = ("NES")

```

Comment:
```suggestion
    system = ("NES")
    patch_suffix = ".apsplunker"

```
Double-check that you didn't mean `apspelunker` too I suppose. I think it's consistent in the world, it's just slightly odd that it's off by a letter.

Reply:
Intended. Spelunker isn’t really a game I could apbbreviate, calling it apspelunker didn’t feel right and I wasn’t comfortable with my other idea of calling it `aplunker`. If this gets significant pushback I’ll change it

Reply:
This was a suggestion plus an extra comment. You should add the `patch_suffix` line to your client so that the launcher registers it to BizHawkClient. It replaces the whole business of importing `components` and modifying the `SuffixIdentifier`.